### PR TITLE
able to get chunk Index from Index, fix writer

### PR DESCRIPTION
**Pull request by helinwang**

### CODE REVIEW

File: chunk.go [hunk 1]
```diff
@@ -76,9 +76,17 @@ func (ch *Chunk) dump(w io.Writer, compressorIndex int) error {
 	return nil
 }
 
+type dummyCloser struct {
```
Comment (wangkuiyi) [inline chunk.go]: dummyCloser ==> noopCompressor ?

File: chunk.go [hunk 2]
```diff
@@ -76,9 +76,17 @@ func (ch *Chunk) dump(w io.Writer, compressorIndex int) error {
...
+	bytes.Buffer
+}
+
+func (c *dummyCloser) Close() error {
+	return nil
+}
+
 func compressData(src io.Reader, compressorIndex int) (*bytes.Buffer, error) {
-	compressed := new(bytes.Buffer)
```
Comment (wangkuiyi) [inline chunk.go]: I think the function should be somehow like the following:

```go
func compressData(src io.Reader, compressorIndex int) (*bytes.Buffer, error) {
        var compressed bytes.Buffer
	var compressor io.WriteCloser

	switch compressorIndex {
	case NoCompression:
		compressor = &noopCompressor{compressed}
	case Snappy:
		compressor = snappy.NewBufferedWriter(compressed)
	case Gzip:
		compressor = gzip.NewWriter(compressed)
	default:
		return nil, fmt.Errorf("Unknown compression algorithm: %d", compressorIndex)
	}

        defer compressor.Close()
	if _, e := io.Copy(compressor, src); e != nil {
		return nil, fmt.Errorf("Failed to compress chunk data: %v", e)
	}

	return &compressed, nil
}
```

The point is that `compressed` is still of type `bytes.Buffer`, and we just create a `noopCompressor` which works like `snappy.Writer` and `gzip.Writer`.

File: header.go [hunk 3]
```diff
@@ -21,6 +19,11 @@ const (
 	Gzip
 )
 
+const (
```
Comment (wangkuiyi) [inline header.go]: Can we merge this `const` group with the previous one?

File: reader.go [hunk 4]
```diff
@@ -19,11 +23,15 @@ func LoadIndex(r io.ReadSeeker) (*Index, error) {
 	for {
 		hdr, e = parseHeader(r)
 		if e != nil {
+			if e != io.EOF {
+				fmt.Println("parse err:", e)
```
Comment (wangkuiyi) [inline reader.go]: Is this snippet for debugging? Should we remove it?

File: writer.go [hunk 5]
```diff
@@ -21,9 +21,14 @@ type Writer struct {
 // using the deflate algorithm given compression level.  Note that
 // level 0 means no compression and -1 means default compression.
 func NewWriter(w io.Writer, maxChunkSize, compressor int) *Writer {
-	if maxChunkSize <= 0 {
+	if maxChunkSize < 0 {
 		maxChunkSize = defaultMaxChunkSize
 	}
+
+	if compressor == -1 {
```
Comment (wangkuiyi) [inline writer.go]: `== -1` ==> `< 0` ?

File: chunk.go (hunk 1 above)
Comment (helinwang) [reply chunk.go]: Done.

File: chunk.go (hunk 2 above)
Comment (helinwang) [reply chunk.go]: Much better, thanks! Done.

File: header.go (hunk 3 above)
Comment (helinwang) [reply header.go]: Done.

File: reader.go (hunk 4 above)
Comment (helinwang) [reply reader.go]: Yes, done.

File: writer.go (hunk 5 above)
Comment (helinwang) [reply writer.go]: Done.

**Pull request merged**